data/transactions/logic/eval_test.go

```diff
@@ -1281,3 +1283,77 @@ func BenchmarkSha512_256(b *testing.B) {
 		require.True(b, pass)
 	}
 }
+
+func TestEd25519verify(t *testing.T) {
+	t.Parallel()
+	var s crypto.Seed
+	crypto.RandBytes(s[:])
+	c := crypto.GenerateSignatureSecrets(s)
+	msg := "62fdfc072182654f163f5f0f9a621d729566c74d0aa413bf009c9800418c19cd"
+	data, err := hex.DecodeString(msg)
+	require.NoError(t, err)
+	pk := basics.Address(c.SignatureVerifier)
+	pkStr := pk.String()
+	program, err := AssembleString(fmt.Sprintf(`arg 0
+arg 1
+addr %s
+ed25519verify`, pkStr))
+	require.NoError(t, err)
+	sig := c.SignBytes(data[:])
+	var txn transactions.SignedTxn
+	txn.Lsig.Logic = program
+	txn.Lsig.Args = [][]byte{data[:], sig[:]}
+	sb := strings.Builder{}
+	ep := EvalParams{Txn: &txn, Trace: &sb}
+	pass, err := Eval(program, ep)
+	if !pass {
+		t.Log(hex.EncodeToString(program))
+		t.Log(sb.String())
+	}
+	require.True(t, pass)
+	require.NoError(t, err)
+
+	// flip a bit and it should not pass
+	msg1 := "52fdfc072182654f163f5f0f9a621d729566c74d0aa413bf009c9800418c19cd"
+	data1, err := hex.DecodeString(msg1)
+	require.NoError(t, err)
+	txn.Lsig.Args = [][]byte{data1, sig[:]}
+	sb1 := strings.Builder{}
+	ep1 := EvalParams{Txn: &txn, Trace: &sb1}
+	pass1, err := Eval(program, ep1)
+	require.False(t, pass1)
+	require.NoError(t, err)
+}
+
+func BenchmarkEd25519Verify(b *testing.B){
+	sb := strings.Builder{}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
```

Comment:
move setup to a separate loop. it should be okay to allocate b.N test [32]byte data values and [64]byte sigs. then run the benchmark in a loop with the timer on for the whole set.

Reply:
done. interestingly, the time is a bit faster.